src/counter.ts

```diff
@@ -33,12 +26,7 @@ export class CounterTs {
       return new Response("Not found", {status: 404});
     }
 
-    // Return `currentValue`. Note that `this.value` may have been
-    // incremented or decremented by a concurrent request when we
-    // yielded the event loop to `await` the `storage.put` above!
-    // That's why we stored the counter value created by this
-    // request in `currentValue` before we used `await`.
-    return new Response(currentValue.toString());
+    return new Response(value.toString());
```

Comment:
```suggestion
    // We don't have to worry about a concurrent request having modified the
    // value in storage because "input gates" will automatically protect against
    // unwanted concurrency. So, read-modify-write is safe. For more details,
    // see: https://blog.cloudflare.com/durable-objects-easy-fast-correct-choose-three/
    await this.state.storage?.put("value", value);
      
    return new Response(value.toString());
```